### PR TITLE
Remove hook latex

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
Restriction of https://github.com/ankidroid/Anki-Android/pull/6268 restricted to LaTeX part for simplicity

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java [hunk 1]
```diff
@@ -35,6 +35,7 @@
 import com.ichi2.compat.CompatHelper;
 import com.ichi2.libanki.exception.NoSuchDeckException;
 import com.ichi2.libanki.hooks.Hooks;
+import com.ichi2.libanki.hooks.ChessFilter;
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java:38]: Codacy Issue found: [Avoid unused imports such as 'com.ichi2.libanki.hooks.ChessFilter'](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5575750)

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5575750&bid=18279991)

File: AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java [hunk 2]
```diff
@@ -1057,7 +1057,9 @@ public void updateFieldCache(long[] nids) {
             }
             fields = (Map<String, String>) Hooks.runFilter("mungeFields", fields, model, data, this);
             String html = new Template(format, fields).render();
-            d.put(type, (String) Hooks.runFilter("mungeQA", html, type, fields, model, data, this));
+            html = (String) Hooks.runFilter("mungeQA", html, type, fields, model, data, this);
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java:1060]: Is this line necessary?

Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java:1060]: Ah, I see, presumably there are other filters which mungeQA still calls?

Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java:1060]: I don't understand this question. What would you want to do instead ? Have a series of imbricated function call ? 
I could do 
```
d.put(type, (String) Hooks.runFilter("mungeQA", LaTeX.mungeQA(html, this, model), type, fields, model, data, this));
```
but that would be horrible to read.
I believe that storing the intermediate result in html is far more readable

Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java:1060]: Yeah. That's the kind of problem that I mentionned earlier. I can't make a clean code and remove hook if I only remove hooks one by one

**Comment by Arthur-Milchior** [comment]
Once merged, i can ensure latex source is shown in browser

**Comment by david-allison-1** [comment]
Thanks Arthur!

https://www.youtube.com/watch?v=rY0WxgSXdEE

**Pull request merged**